clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/bitbucket/BitbucketArtifactCredentials.java

```diff
@@ -17,19 +17,20 @@
 
 package com.netflix.spinnaker.clouddriver.artifacts.bitbucket;
 
+import com.google.common.collect.ImmutableList;
 import com.netflix.spinnaker.clouddriver.artifacts.config.ArtifactCredentials;
 import com.netflix.spinnaker.clouddriver.artifacts.config.SimpleHttpArtifactCredentials;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
 import com.squareup.okhttp.OkHttpClient;
-import java.util.Collections;
-import java.util.List;
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 
+@NonnullByDefault
 @Slf4j
 final class BitbucketArtifactCredentials
     extends SimpleHttpArtifactCredentials<BitbucketArtifactAccount> implements ArtifactCredentials {
   @Getter private final String name;
-  @Getter private final List<String> types = Collections.singletonList("bitbucket/file");
+  @Getter private final ImmutableList<String> types = ImmutableList.of("bitbucket/file");
```

Comment:
This doesn't provide any benefit, but adds a guava dependency to this class (Collections.singletonList is already an immutable list, is there any strong reason to expose the details of the collection type?

Reply:
I guess the main benefit is that returning a `List` that happens to be immutable doesn't give any hint to the consumer that it is immutable.  This makes it clear to the consumer that the returned list is immutable, which seems like a reasonable detail to expose.  I know @plumpy also has thoughts on this, so interested in his opinion.